lib/puppetfile-resolver/spec_searchers/forge.rb

```diff
@@ -52,11 +55,27 @@ def self.fetch_all_module_releases(owner, name, forge_api_url = DEFAULT_FORGE_UR
           http_options[:ca_file] = PuppetfileResolver::Util.static_ca_cert_file if ENV['SSL_CERT_FILE'].nil?
 
           response = nil
-          Net::HTTP.start(uri.host, uri.port, http_options) do |http|
-            request = Net::HTTP::Get.new uri
-            response = http.request request
+
+          begin
+            ::PuppetfileResolver::Util.with_proxy(forge_proxy) do
+              Net::HTTP.start(uri.host, uri.port, http_options) do |http|
+                request = Net::HTTP::Get.new uri
+                response = http.request request
+              end
+            end
+          rescue => e
```

Comment:
This is probably dangerous given there's not typing on `e` here so it'll trap everything and anything

Reply:
I'll update to a more specific `rescue SocketError`.